packages/wallet/src/redux/protocols/advance-channel/readme.md

```diff
@@ -43,29 +43,37 @@ It can also happen if the `commitmentType` is incompatible with the latest commi
 
 - To simplify the communication assumptions in a channel with an arbitrary number of participants, each participant sends a valid round of commitments only to the next participant.
 
+## Open questions
+
+- What should cause the protocol to fail?
+
 ## State machine diagram
 
 ```mermaid
 graph TD
 linkStyle default interpolate basis
   St((start)) --> ICO{Is Channel Open}
-  ICO --> |Yes| NSTS(NotSafeToSend)
+  ICO --> |Yes| STSCO{Safe to send?}
+  STSCO --> |Yes| CS
+  STSCO --> |No| NSTS(NotSafeToSend)
   ICO --> |No| FP{First participant?}
-  FP --> |Yes|CS(CommitmentSent)
+  FP --> |Yes|STS
   FP --> |No| CU(ChannelUnknown)
-  RC   -->|No| CS
-  CS   -->|CommitmentReceived| RC
-  CU   -->|CommitmentReceived| RC{Round Complete?}
+  CS   -->|CommitmentReceived| RC{Round Complete?}
   NSTS -->|CommitmentReceived| STS{Safe to send?}
+  CU   -->|CommitmentReceived| STS
+  CU   -->|ClearedToSend| STS
```

Comment:
Is there a scenario for `ChannelUnknown` -> `Safe to Send` -> `ChannelUnknown`?

Reply:
Added three tests to cover this action.